src/Nethermind/Nethermind.Consensus.AuRa/AuRaBlockProducer.cs

```diff
@@ -125,5 +128,40 @@ protected override Task<Block> SealBlock(Block block, BlockHeader parent, Cancel
             _reportingValidator.TryReportSkipped(block.Header, parent);
             return base.SealBlock(block, parent, token);
         }
+        
+        private class ValidateTxSource : ITxSource
+        {
+            private readonly ITxSource _innerSource;
+            private readonly ILogger _logger;
+
+            public ValidateTxSource(ITxSource innerSource, ILogManager logManager)
+            {
+                _innerSource = innerSource;
+                _logger = logManager.GetClassLogger();
+                if (_logger.IsInfo) _logger.Info($"Transaction sources used when building blocks: {this}");
+            }
+
+            public IEnumerable<Transaction> GetTransactions(BlockHeader parent, long gasLimit)
+            {
+                IDictionary<(Address, UInt256), Keccak> senderNonces = new Dictionary<(Address, UInt256), Keccak>();
```

Comment:
new Dictionary on each request?
can you reuse the same one via Resettables?

Reply:
its 1 small dictionary per 5 sec